pandapipes/component_models/abstract_models/branch_models.py

```diff
@@ -282,78 +272,5 @@ def calculate_temperature_lift(cls, net, branch_pit, node_pit):
-        to_junction_nodes = node_active_idx_lookup[junction_idx_lookup[
-            net[cls.table_name()]["to_junction"].values[placement_table]]]
-
-        from_nodes = branch_pit[:, FROM_NODE].astype(np.int32)
-        to_nodes = branch_pit[:, TO_NODE].astype(np.int32)
-        fluid = get_fluid(net)
-
-        v_mps = branch_pit[:, VINIT]
-
-        t0 = node_pit[from_nodes, TINIT_NODE]
-        t1 = node_pit[to_nodes, TINIT_NODE]
-        mf = branch_pit[:, LOAD_VEC_NODES]
-        vf = branch_pit[:, LOAD_VEC_NODES] / get_fluid(net).get_density((t0 + t1) / 2)
-
-        idx_active = branch_pit[:, ELEMENT_IDX]
-        _, v_sum, mf_sum, vf_sum, internal_pipes = _sum_by_group(idx_active, v_mps, mf, vf, np.ones_like(idx_active))
-
-        if fluid.is_gas:
-            # derived from the ideal gas law
-            p_from = node_pit[from_nodes, PAMB] + node_pit[from_nodes, PINIT]
-            p_to = node_pit[to_nodes, PAMB] + node_pit[to_nodes, PINIT]
-            numerator = NORMAL_PRESSURE * branch_pit[:, TINIT]
-            normfactor_from = numerator * fluid.get_property("compressibility", p_from) \
-                              / (p_from * NORMAL_TEMPERATURE)
-            normfactor_to = numerator * fluid.get_property("compressibility", p_to) \
-                            / (p_to * NORMAL_TEMPERATURE)
-            v_gas_from = v_mps * normfactor_from
-            v_gas_to = v_mps * normfactor_to
-
-            _, nf_from_sum, nf_to_sum = _sum_by_group(idx_active, normfactor_from, normfactor_to)
-
-            v_gas_from_ordered = select_from_pit(from_nodes, from_junction_nodes, v_gas_from)
-            v_gas_to_ordered = select_from_pit(to_nodes, to_junction_nodes, v_gas_to)
-
-            res_table["v_from_m_per_s"].values[placement_table] = v_gas_from_ordered
-            res_table["v_to_m_per_s"].values[placement_table] = v_gas_to_ordered
-            res_table["normfactor_from"].values[placement_table] = nf_from_sum / internal_pipes
-            res_table["normfactor_to"].values[placement_table] = nf_to_sum / internal_pipes
-
-        res_table["p_from_bar"].values[placement_table] = node_pit[from_junction_nodes, PINIT]
-        res_table["p_to_bar"].values[placement_table] = node_pit[to_junction_nodes, PINIT]
-        res_table["t_from_k"].values[placement_table] = node_pit[from_junction_nodes, TINIT_NODE]
-        res_table["t_to_k"].values[placement_table] = node_pit[to_junction_nodes, TINIT_NODE]
-        res_table["mdot_to_kg_per_s"].values[placement_table] = -mf_sum / internal_pipes
-        res_table["mdot_from_kg_per_s"].values[placement_table] = mf_sum / internal_pipes
-        res_table["vdot_norm_m3_per_s"].values[placement_table] = vf_sum / internal_pipes
-        return placement_table, res_table, branch_pit, node_pit
+    def extract_results(cls, net, options, branch_results, nodes_connected, branches_connected):
+        raise NotImplementedError
```

Comment:
The problem is that raise NotImplementedError makes a class an abstract class (the method is not defined). If we don't do this in the subclass, it becomes an abstract class without anyone really noticing it. Another option would be to use ABC (AbstractBaseClass) and make them inherit from it, but I am not sure, if this could actually mess up with other features we implemented (like JSON serialization, classmethods, ... ). It's just for being more concise in our code.

Reply:
I would leave it like that at the moment, but it could still be adapted later on.